make.ps1

```diff
@@ -0,0 +1,138 @@
+
+$_BASEDIR = Split-Path -Path $MyInvocation.MyCommand.Definition -Parent
+
+$_VSWHERE = [System.IO.Path]::Combine(${env:ProgramFiles(x86)}, 'Microsoft Visual Studio\Installer\vswhere.exe')
+$_VSINSTPATH = ''
+
+if([System.IO.File]::Exists($_VSWHERE)) {
+    $_VSINSTPATH = & "$_VSWHERE" -latest -requires Microsoft.Component.MSBuild -property installationPath
+} else {
+    Write-Error "Visual Studio 2017 15.2 or later is required"
+    Exit 1
+}
+
+if([System.IO.File]::Exists([System.IO.Path]::Combine($_VSINSTPATH, 'MSBuild\15.0\Bin\MSBuild.exe'))) {
+    $env:PATH = [String]::Join(';', $env:PATH, [System.IO.Path]::Combine($_VSINSTPATH, 'MSBuild\15.0\Bin'))
+}
+
+# Details for running tests on the various frameworks
+$_FRAMEWORKS = @{
+    "net40" = @{
+        "runner" = [System.IO.Path]::Combine($_BASEDIR, 'packages\nunit.consolerunner\3.7.0\tools\nunit3-console.exe');
+        "args" = @("--params", '"FRAMEWORK=__FRAMEWORK__"', '--labels=All', '--result:__FILTERNAME__-__FRAMEWORK__-__CONFIGURATION__-result.xml', '__BASEDIR__\bin\__CONFIGURATION__\__FRAMEWORK__\IronPythonTest.dll' );        
+        "filterArg" = '--where:"__FILTER__"';
+        "filters" = @{
+            "all" = "";
+            "smoke" = "Category==StandardCPython";
+            "cpython" = "Category==StandardCPython || Category==AllCPython";
+            "ironpython" = "Category==IronPython";
+        }
+    };
+    "net45" = @{
+        "runner" = "dotnet";
+        "args" = @('test', '__BASEDIR__\Src\IronPythonTest\IronPythonTest.csproj', '-f', '__FRAMEWORK__', '-o', '__BASEDIR__\bin\__CONFIGURATION__\__FRAMEWORK__', '-c', '__CONFIGURATION__', '--no-build', '-l', "trx;LogFileName=__FILTERNAME__-__FRAMEWORK__-__CONFIGURATION__-result.trx", '-s', 'runsettings.__FRAMEWORK__');
+        "filterArg" = '--filter="__FILTER__"';
+        "filters" = @{
+            "all" = "";
+            "smoke" = "TestCategory=StandardCPython";
+            "cpython" = "TestCategory=StandardCPython | TestCategory=AllCPython";
+            "ironpython" = "TestCategory=IronPython";
+        }
+    };
+    "netcoreapp2.0" = @{
+        "runner" = "dotnet";
+        "args" = @('test', '__BASEDIR__\Src\IronPythonTest\IronPythonTest.csproj', '-f', '__FRAMEWORK__', '-o', '__BASEDIR__\bin\__CONFIGURATION__\__FRAMEWORK__', '-c', '__CONFIGURATION__', '--no-build', '-l', "trx;LogFileName=__FILTERNAME__-__FRAMEWORK__-__CONFIGURATION__-result.trx", '-s', 'runsettings.__FRAMEWORK__');
+        "filterArg" = '--filter="__FILTER__"';
+        "filters" = @{
+            "all" = "";
+            "smoke" = "TestCategory=StandardCPython";
+            "cpython" = "TestCategory=StandardCPython | TestCategory=AllCPython";
+            "ironpython" = "TestCategory=IronPython";
+        }
+    }
+}
+
+function Main([String] $target, [String] $configuration) {
+    msbuild Build.proj /t:$target /p:BuildFlavour=$configuration /verbosity:minimal /nologo /p:Platform="Any CPU" /bl:build-$target.binlog
```

Comment:
Should this have /m parameter as well?

Reply:
It historically has not, but we could add it

Reply:
Does our build process work properly in parallel? If so then might as well add it.

Reply:
I built locally and it didn't die, so I'll add it